system/src/commonTest/kotlin/com/splendo/kaluga/system/network/NetworkStateTest.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import com.splendo.kaluga.flow.Flowable
+import com.splendo.kaluga.test.FlowTestBlock
+import com.splendo.kaluga.test.FlowableTest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class NetworkStateTest : FlowableTest<NetworkState>() {
+
+    lateinit var networkStateRepo: NetworkStateRepo
+    lateinit var networkManager: MockNetworkManager
+
+    override fun flowable(): Flowable<NetworkState> = networkStateRepo.flowable
+
+    private val testCoroutine = SupervisorJob()
+
+    private val coroutineScope = CoroutineScope(Dispatchers.Main + testCoroutine)
```

Comment:
I know it's copied from somewhere but you don't seem to use it for anything (like testing cancelation)

Reply:
Do you mean to remove `testCoroutine` or `coroutineScope`? `NetworkStateRepo` take it as a parameter to give it to the parent class `ColdStateRepo`

Reply:
Everything you copied that is not needed, jobs, scopes, etc. The test has defaults for these.
You don't actually use the scope either, just the context, but it's not clear what for (ColdStateRepo has a default value for this).

Reply:
3901103

Reply:
Thanks, after reading more carefully what you said [here](https://github.com/splendo/kaluga/pull/197#discussion_r521671006), I realized what you meant and the work for this change is in progress.